src/js/facility-locator/components/Listbox.jsx

```diff
@@ -0,0 +1,161 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import { truncate, kebabCase } from 'lodash';
+import { benefitsServices, facilityTypes, vetCenterServices } from '../config';
+import { getServices } from '../utils/helpers.js';
+
+class Listbox extends Component {
+
+  isSelectedOption = (option, type) => {
+    return this.props.currentQuery[type] === option;
+  }
+
+  renderSelectOptions = () => {
+    const { currentQuery: { facilityType, serviceType } } = this.props;
+    const services = getServices(facilityType, benefitsServices, vetCenterServices);
+    this.props.resetOptions();
+    return (
+      <ul
+        className="dropdown"
+        role={!this.props.isDisabled ? 'listbox' : undefined}
+        id="service-list">
+        {services && services.map((k, i) => {
+          const defaultSelected = !serviceType && k === 'All';
+          const isSelected = this.isSelectedOption(k, 'serviceType');
+          const isHovered = defaultSelected || isSelected;
+          const serviceOptionClasses = classNames({
+            'dropdown-option': true,
+            'facility-option': true,
+            'is-hovered': isHovered
+          });
+
+          return (
+            <li
+              tabIndex={this.props.dropdownActive ? '0' : '-1'}
+              role="option"
+              aria-selected={isHovered}
+              className={serviceOptionClasses}
+              key={k}
+              value={k}
+              ref={ elem => { this.props.options[i] = elem; }}
+              id={k}
+              onClick={() => this.props.handleFilterSelect(k, 'service')}>
+              <span className="flex-center">
+                <span className="legend spacer"></span>
+                {benefitsServices[k] || k}
+              </span>
+            </li>
+          );
+        })
+        }
+      </ul>
+    );
+  }
+
+  renderSelectOptionWithIcon = (facilityType, index) => {
+    const defaultSelected = !this.props.currentQuery.facilityType && !facilityType;
+    const isSelected = this.isSelectedOption((facilityType), 'facilityType');
+    const isHovered = isSelected || defaultSelected;
+    const facilityOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true,
+      'is-hovered': isHovered
+    });
+    const facilityIconClasses = classNames({
+      legend: true,
+      spacer: !facilityType,
+      [`${kebabCase(facilityType)}-icon`]: facilityType
+    });
+
+    return (
+      <li
+        role="option"
+        tabIndex={this.props.dropdownActive ? '0' : '-1'}
+        aria-selected={isHovered}
+        ref={ elem => { this.props.options[index] = elem; }}
+        id={index > -1 ? (facilityType || 'AllFacilities') : null}
```

Comment:
It looks like index is always greater than -1.

Reply:
Good catch.